packages/core-js/modules/es.string.replace.js

```diff
@@ -112,12 +112,12 @@ require('../internals/fix-regexp-well-known-symbol-logic')(
             break;
           default: // \d\d?
             var n = +ch;
-            if (n === 0) return ch;
+            if (n === 0) return match;
```

Comment:
When the replacement is, for example `foo $12 bar`, `ch` is `12` while `match` is `$12`.